Hardware/Hardware/CardReader/UnderlyingError.swift

```diff
@@ -0,0 +1,308 @@
+    /// - The user’s device is in Airplane Mode and unable to gather location data.
+    case locationServicesDisabled
+
+    /// This error indicates that Bluetooth is turned off, and the user should use Settings to turn Bluetooth on.
+    /// If Bluetooth is on but the app does not have permission to use it, a different error (bluetoothError) occurs.
+    case bluetoothDisabled
+
+    /// Generic bluetooth error. Among other things, it may indicate that the app does not have permission to use Bluetooth.
+    case bluetoothError
+
+    /// Scanning for bluetooth devices timed out.
+    case bluetoothScanTimedOut
+
+    /// Bluetooth Low Energy is unsupported on this iOS device. Use a different iOS device that supports BLE (also known as Bluetooth 4.0)
+    case bluetoothLowEnergyUnsupported
+
+    /// Updating the reader software failed because the reader’s battery is too low. Charge the reader before trying again.
+    case readerSoftwareUpdateFailedBatteryLow
+
+    /// Updating the reader software failed because the update was interrupted.
+    case readerSoftwareUpdateFailedInterrupted
+
+    /// Generic reader software update error.
+    case readerSoftwareUpdateFailed
+
+    /// Updating the reader software failed because there was an error communicating with the reader.
+    case readerSoftwareUpdateFailedReader
+
+    /// Updating the reader software failed because there was an error communicating with the update server.
+    case readerSoftwareUpdateFailedServer
+
+    /// The card is not a chip card.
+    case cardInsertNotRead
+
+    /// The swipe could not be read.
+    case cardSwipeNotRead
+
+    /// Reading a card timed out.
+    case cardReadTimeOut
+
+    /// The card was removed during the transaction.
+    case cardRemoved
+
+    /// A card can only be used for one transaction, and must be removed after being read
+    case cardLeftInReader
+
+    /// The reader is busy.
+    case readerBusy
+
+    /// An incompatible reader was detected.
+    case readerIncompatible
+
+    /// Could not communicate with the reader.
+    case readerCommunicationError
+
+    /// Connecting to the bluetooth device timed out.
+    /// Make sure the device is powered on, in range, and not connected to another app or device.
+    /// If this error continues to occur, you may need to charge the device.
+    case bluetoothConnectTimedOut
+
+    /// The Bluetooth device was disconnected unexpectedly.
+    case bluetoothDisconnected
+
+    /// An attempt to process a payment was made from a reader with an unsupported reader version.
+    /// You will need to update your reader to the most recent version in order to accept payments
+    case unsupportedReaderVersion
+
+    /// Connecting to the reader failed because it is currently in use
+    case connectFailedReaderIsInUse
+
+    /// Call 911. Unexpected SDK error.
+    case unexpectedSDKError
+
+    /// The Stripe API declined the payment
+    case paymentDeclinedByPaymentProcessorAPI
```

Comment:
Let me know if you have any trouble resolving the likely merge conflict here with https://github.com/woocommerce/woocommerce-ios/pull/5511